Marlin/Marlin_main.cpp

```diff
@@ -8281,29 +8271,33 @@ inline void gcode_M907() {
 
 #if HAS_CASE_LIGHT
 
-  uint8_t case_light_brightness = 255;
+  uint8_t case_light_brightness;
 
   void update_case_light() {
-    digitalWrite(CASE_LIGHT_PIN, case_light_on != INVERT_CASE_LIGHT ? HIGH : LOW);
-    analogWrite(CASE_LIGHT_PIN, case_light_on != INVERT_CASE_LIGHT ? case_light_brightness : 0);
+    digitalWrite(CASE_LIGHT_PIN, (case_light_brightness >= 128) != INVERT_CASE_LIGHT ? HIGH : LOW);
+    analogWrite(CASE_LIGHT_PIN, INVERT_CASE_LIGHT ? 255 - case_light_brightness : case_light_brightness );
```

Comment:
I think the light toggle should only change `case_light_on`, and then this code should be:
```cpp
const uint8_t clb = case_light_on ? case_light_brightness : (INVERT_CASE_LIGHT ? 255 : 0);
digitalWrite(CASE_LIGHT_PIN, (clb >= 128) != INVERT_CASE_LIGHT ? HIGH : LOW);
analogWrite(CASE_LIGHT_PIN, INVERT_CASE_LIGHT ? 255 - clb : clb);
```